code/go/0chain.net/smartcontract/minersc/benchmark_rest_tests.go

```diff
@@ -1,126 +1,159 @@
 package minersc
 
 import (
-	"context"
-	"net/url"
+	"0chain.net/smartcontract/stakepool/spenum"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/require"
+	"io"
+	"net/http"
+	"net/http/httptest"
 	"testing"
 
 	cstate "0chain.net/chaincore/chain/state"
-	"0chain.net/chaincore/smartcontract"
-	sci "0chain.net/chaincore/smartcontractinterface"
 	"0chain.net/chaincore/transaction"
 	bk "0chain.net/smartcontract/benchmark"
-	"0chain.net/smartcontract/stakepool/spenum"
 )
 
 type RestBenchTest struct {
-	name     string
-	endpoint func(
-		context.Context,
-		url.Values,
-		cstate.StateContextI,
-	) (interface{}, error)
-	params url.Values
+	name        string
+	params      map[string]string
+	shownResult bool
 }
 
-func (rbt RestBenchTest) Name() string {
-	return rbt.name
+func (rbt *RestBenchTest) Name() string {
+	return "miner_rest." + rbt.name
 }
 
-func (rbt RestBenchTest) Transaction() *transaction.Transaction {
+func (rbt *RestBenchTest) Transaction() *transaction.Transaction {
 	return &transaction.Transaction{}
 }
 
-func (rbt RestBenchTest) Run(balances cstate.StateContextI, _ *testing.B) error {
-	_, err := rbt.endpoint(context.TODO(), rbt.params, balances)
-	return err
+func (rbt *RestBenchTest) Run(balances cstate.StateContextI, b *testing.B) error {
```

Comment:
looks like this code is the same for different benchmarks, it can be moved to common package